scripts/globals/summon.lua

```diff
@@ -1,241 +1,187 @@
-function avatarFSTR(atk_str,def_vit)
-    local dSTR = atk_str - def_vit;
-    if (dSTR >= 12) then
-        fSTR2 = ((dSTR+4)/2);
-    elseif (dSTR >= 6) then
-        fSTR2 = ((dSTR+6)/2);
-    elseif (dSTR >= 1) then
-        fSTR2 = ((dSTR+7)/2);
-    elseif (dSTR >= -2) then
-        fSTR2 = ((dSTR+8)/2);
-    elseif (dSTR >= -7) then
-        fSTR2 = ((dSTR+9)/2);
-    elseif (dSTR >= -15) then
-        fSTR2 = ((dSTR+10)/2);
-    elseif (dSTR >= -21) then
-        fSTR2 = ((dSTR+12)/2);
-    else
-        fSTR2 = ((dSTR+13)/2);
+    -- apply ftp bonus
+    if (tpeffect == TP_DMG_BONUS) then
+        finaldmg = finaldmg * avatarFTP(skill:getTP(), mtp100, mtp200, mtp300)
     end
-    -- Apply fSTR caps.
-    if (fSTR2< -1) then
-        fSTR2 = -1;
-    elseif (fSTR2>8) then
-        fSTR2 = 8;
+
+    returninfo.dmg = finaldmg
+    returninfo.hitslanded = hitslanded
+
+    return returninfo
+end
+
+-- minFstr = dSTR/4 + 0.5
+-- maxFstr = dSTR/4 + 0.25
+function avatarFSTR(str, def_vit)
+    local dSTR = atk_str - def_vit
+    return math.floor(dSTR/4 + 0.5), math.floor(dSTR/4 + 0.25)
+end
+
+function avatarHitDmg(dmg, pdifMin, pdifMax, fstrMin, fstrMax, critrate)
+    local fstr = math.random(fstrMin, fstrMax)
+    local pdif = math.random(pdifMin * 1000, pdifMax * 1000) / 1000
+    if math.random() <= critrate then
+        pdif = math.min(pdif + 1, 4.2)
     end
-    return fSTR2;
-end;
+    return (dmg + fstr) * pdif
+end
 
 function AvatarFinalAdjustments(dmg,mob,skill,target,skilltype,skillparam,shadowbehav)
 
     -- physical attack missed, skip rest
     if (skilltype == dsp.attackType.PHYSICAL and dmg == 0) then
-        return 0;
+        return 0
     end
 
     -- set message to damage
     -- this is for AoE because its only set once
-    skill:setMsg(dsp.msg.basic.DAMAGE);
+    skill:setMsg(dsp.msg.basic.DAMAGE)
 
     --Handle shadows depending on shadow behaviour / skilltype
     if (shadowbehav < 5 and shadowbehav ~= MOBPARAM_IGNORE_SHADOWS) then --remove 'shadowbehav' shadows.
-        targShadows = target:getMod(dsp.mod.UTSUSEMI);
-        shadowType = dsp.mod.UTSUSEMI;
+        targShadows = target:getMod(dsp.mod.UTSUSEMI)
+        shadowType = dsp.mod.UTSUSEMI
         if (targShadows==0) then --try blink, as utsusemi always overwrites blink this is okay
-            targShadows = target:getMod(dsp.mod.BLINK);
-            shadowType = dsp.mod.BLINK;
+            targShadows = target:getMod(dsp.mod.BLINK)
+            shadowType = dsp.mod.BLINK
         end
 
         if (targShadows>0) then
         -- Blink has a VERY high chance of blocking tp moves, so im assuming its 100% because its easier!
             if (targShadows >= shadowbehav) then --no damage, just suck the shadows
-                skill:setMsg(dsp.msg.basic.SHADOW_ABSORB);
-                target:setMod(shadowType,(targShadows-shadowbehav));
+                skill:setMsg(dsp.msg.basic.SHADOW_ABSORB)
```

Comment:
Gut parenth in all these conditionals
Don't need to surround the math in parentheses either (I know that was there before and wasn't you)